### ISSUE TITLE
Misinterpreting actual page numbers to file's page numbers

### ISSUE THREAD

**Issue by autoditactics** [kcroker/dpsprep#4]
I have a DJVU file whose conversion has incorrect page numbers in its TOC are wrong.
bmarks.out:
> (bookmarks
 ("Cover"
  "#cover_2.djvu" )
 ("Front matter"
  "#_005.djvu" )
 ("Preface to the Dover Edition "
  "#_008.djvu" )
 ("Preface "
  "#_010.djvu" )
 ("Table of Contents "
  "#_012.djvu" )
 ("Introduction"
  "#_014.djvu" )
 ("Chapter 1. The Basic Notions "
  "#003.djvu" )
 ("Chapter 2. Order and Well-Foundedness "
  "#032.djvu" )
 ("Chapter 3. Cardinal Numbers"
  "#076.djvu" )
 ("Chapter 4. The Ordinals"
  "#112.djvu" )
 ("Chapter 5. The Axiom of Choice and Some of its Consequences"
  "#158.djvu" )
 ("Chapter 6. A Review of Point Set Topology. "
  "#199.djvu" )
 ("Chapter 7. The Real Spaces "
  "#216.djvu" )
 ("Chapter 8. Boolean Algebras "
  "#244.djvu" )
 ("Chapter 9. Infinite Combinatorics and Large Cardinals "
  "#289.djvu" )
 ("Appendix 10. The Eliminability and Conservation Theorems"
  "#357.djvu" )
 ("Bibliography"
  "#367.djvu" )
 ("Additional Bibliography "
  "#376.djvu" )
 ("Index of Notation "
  "#377.djvu" )
 ("Index"
  "#383.djvu" )
 ("Appendix: Corrections and Additions "
  "#393.djvu" ) )

These numbers represent the actual page numbers (as opposed to the file's page numbers), and when converted to PDF, it misinterprets these numbers as the file's page numbers, with the exception that the page numbers that have an underscore seem to be totally ignored (they seem to represent "negative" page numbers relative to page 1). The DJVU file can be found here: https://drive.google.com/open?id=1zVKe0qXA08_q5Tq42LzvNVVgRXOeHVyM

There is probably a way of determining where page 1 of the actual book/article/etc. is since DjView is able to display everything correctly.

**Comment by kcroker** [comment]
Thank you for using the script I wrote, I'm glad you've (almost) found it
useful :D

It should have gotten the internal document page numbers correctly, I
think?  I've not had the problem, and others have not reported an issue in
a long time.  There was one outstanding bug report of something like this,
did you see it on github?  Did you try it with a separate DJVU file, to see
if there is still trouble?

I've got the drive link to your DJVU file, so I will be able to see if I
can reproduce the issue.  I've not altered the dpsprep package in maybe 2
years, so I'll have to poke around a bit.  I'm pretty swamped at the moment
with work stuff though, so I won't be able to look for a bit.  If you find
the problem and patch it, I will definitely test and accept any pull
request!

Word is bond.
-k

**Comment by Okoyos** [comment]
Hi,
I have found your script very useful (though only recently as pdftk was giving me problems). The link in the other open issue seems to be dead, so I can't confirm whether or not it is the same issue. The other files I've tried don't have these kind of page numbers with the .djvu extension or with underscores. They're just normal integers, and those work fine.

One other small note, the directory of the output pdf is relative to the temporary .dsprep directory instead of the directory of the input djvu (eg. "~/Desktop/out.pdf" is interpreted as "~/.dpsprep/Users/<user>/Desktop/out.pdf"). It was a minor annoyance in the beginning.

I'm not that great at python, so I haven't considered writing a pull request, though it is possible.
